src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -221,7 +242,8 @@ public bool ExecuteReplicationOnce(OutgoingReplicationStatsScope stats)
                         // the last etag they have from us on the other side
                         _parent._lastSentDocumentEtag = _lastEtag;
                         _parent._lastDocumentSentTime = DateTime.UtcNow;
-                        _parent.SendHeartbeat(DocumentsStorage.GetDatabaseChangeVector(documentsContext));
+                        var changeVector = wasDelayed ? null : DocumentsStorage.GetDatabaseChangeVector(documentsContext);
+                        _parent.SendHeartbeat(changeVector);
```

Comment:
Shouldn't we send the last change vector we scanned here? (Up to the item we stopped for being too early)

Reply:
If we have documents to send, but were delayed (in this particular case we have _nothing_ to send), we will be sending the current change vector without sending the actual documents.

Reply:
What if we filtered documents and still have things to send?

Reply:
So we are not ending in this code path. This is only if we have nothing to send